File: WorkSphere.Server/Model/ProjectManager.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using WorkSphere.Model;

namespace WorkSphere.Server.Model
{
    public class ProjectManager : BaseEntity
    {
        [Required]
        [StringLength(50, MinimumLength = 2, ErrorMessage = "First Name must be between 2 and 50 characters")]
        public string FirstName { get; set; } = string.Empty;

        [Required]
        [StringLength(50, MinimumLength = 2, ErrorMessage = "Last Name must be between 2 and 50 characters")]
        public string LastName { get; set; } = string.Empty;

        [Required]
        [EmailAddress(ErrorMessage = "Invalid Email Address")]
        public string Email { get; set; } = string.Empty;

        [Required]
        [DataType(DataType.Date)]
        public DateTime EmploymentDate { get; set; }

        [NotMapped]
        public string FullName => $"{FirstName} {LastName}";

        // One-to-Many: A ProjectManager manages multiple projects
        [JsonIgnore]
        public virtual ICollection<Project>? ManagedProjects { get; set; }

        // One-to-One: ProjectManager has a salary
        public int? SalaryID { get; set; }
        public virtual Salary? Salary { get; set; }
    }

}

File: WorkSphere.Client/src/app/features/project-manager/model/project-manager.ts
import {Project} from '../../project/models/project';
import {Salary} from '../../salary/model/salary';

export class ProjectManager {
  id?: number;
  firstName: string;
  lastName: string;
  email: string;
  employmentDate: Date | string;
  salaryID?: number;
  salary?: Salary;
  managedProjects: Project[];

  constructor(id: number = null, firstName: string, lastName: string, email: string, employmentDate: Date, salaryID?: number, salary?: Salary, managedProjects: Project[] = []) {
    this.id = id;
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
    this.employmentDate = employmentDate;
    this.salaryID = salaryID;
    this.salary = salary;
    this.managedProjects = managedProjects;

  }
}
